### DISCUSSION TITLE
RAG Task instruction prefixes - where do they belong

### DISCUSSION THREAD

**Discussion by ghost** [open-webui/open-webui#7996] [Q&A]
For the [nomic-embed-text-v1.5](https://huggingface.co/nomic-ai/nomic-embed-text-v1.5#task-instruction-prefixes) there are Instruction prefixes almost mandatory.

The FAQ says:

> Usage
> 
> Important: the text prompt must include a task instruction prefix, instructing the model which task is being performed.
> 
> For example, if you are implementing a RAG application, you embed your documents as search_document: < text here > and embed your user queries as search_query: < text here >.

But where in the flow with prompting (with help of #) the Documents (RAG) does that belong?

1) in the RAG template 
a) ### Task: **HERE:**
b) <user_query>
**HERE:**{{QUERY}}

or

2) in the prompt saved for RAG and when '/' it should be the first word: **HERE:** prompt X content

**Comment by stevenmoto** [comment]
I would like some clarification as well, I am not that technical; in my case I am using VLLM to serve a qwen3 reranker and pointing in OpenWebUI.

Offline and Online inference have vastly different results and must I format the query and document for better reranker results.

For example 
https://github.com/vllm-project/vllm/blob/main/examples/offline_inference/pooling/qwen3_reranker.py

prefix = '<|im_start|>system\nJudge whether the Document meets the requirements based on the Query and the Instruct provided. Note that the answer can only be "yes" or "no".<|im_end|>\n<|im_start|>user\n'
suffix = "<|im_end|>\n<|im_start|>assistant\n<think>\n\n</think>\n\n"

query_template = "{prefix}<Instruct>: {instruction}\n<Query>: {query}\n"
document_template = "<Document>: {doc}{suffix}"

vllm does not format the request and it must be done at the client pre-processing level.
Where in OpenWebUI would I be able to do this via the RAG pipeline?

Any help is appreciated.
